src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int ARRAY_SIZE = 16;
+    private static final int MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[ARRAY_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        int index = hash & table.length - 1;
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value, hash, null);
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node : table) {
```

Comment:
The variable `newNode` is unnecessary and can lead to confusion. You can directly use the `node` variable in the while loop.